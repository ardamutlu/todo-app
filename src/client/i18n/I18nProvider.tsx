import React from 'react';
import { useSelector } from 'react-redux';
import { IntlProvider } from 'react-intl';
import '@formatjs/intl-relativetimeformat/polyfill';
import '@formatjs/intl-relativetimeformat/dist/locale-data/en';
import '@formatjs/intl-relativetimeformat/dist/locale-data/tr';

import enMessages from './messages/en';
import trMessages from './messages/tr';

const allMessages = {
  en: enMessages,
  tr: trMessages
};

export default function I18nProvider({ children }: any) {
  const locale = useSelector(({ i18n }: any) => i18n.lang);
  const messages = allMessages[locale];

  return (
    <IntlProvider locale={locale} messages={messages}>
      {children}
    </IntlProvider>
  );
}
