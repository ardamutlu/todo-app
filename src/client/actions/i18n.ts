import { State } from '../reducers/i18n';

export type Actions = ReturnType<typeof setLanguage>;

export const SET_LANGUAGE = 'SET_LANGUAGE';

export const setLanguage = (lang: State['lang']) =>
  ({
    type: SET_LANGUAGE,
    payload: {
      lang
    }
  } as const);
