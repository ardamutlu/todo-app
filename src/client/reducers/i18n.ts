import * as Actions from '../actions/i18n';

export type State = {
  lang: string;
};

export const initialState: State = {
  lang: 'en'
};

export const reducer = (state: State = initialState, action: Actions.Actions): State => {
  switch (action.type) {
    case Actions.SET_LANGUAGE:
      localStorage.setItem('lang', action.payload.lang);
      return { ...state, lang: action.payload.lang };

    default:
      return state;
  }
};
