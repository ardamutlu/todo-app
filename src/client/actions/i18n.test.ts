import * as actions from './i18n';

test('should create an action to set language', () => {
  expect(actions.setLanguage('en')).toMatchSnapshot();
});
