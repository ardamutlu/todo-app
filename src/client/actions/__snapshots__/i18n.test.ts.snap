// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create an action to set language 1`] = `
Object {
  "payload": Object {
    "lang": "en",
  },
  "type": "SET_LANGUAGE",
}
`;
